csharp/FAudio.cs

```diff
@@ -39,15 +41,30 @@ public static class FAudio
 
 	#region UTF8 Marshaling
 
-	private static byte[] UTF8_ToNative(string s)
+	private static int Utf8Size(string str)
+	{
+		return (str.Length + 1) * 4;
+	}
+
+	private static unsafe byte* Utf8Encode(string str, byte* buffer, int bufferSize)
 	{
-		if (s == null)
+		int strBufferSize = str.Length + 1;
+		char* strBuffer = stackalloc char[strBufferSize];
```

Comment:
Does this work instead of an allocation here? Can verify that this works with VS2010:

```
fixed (char* strPtr = str)
{
    Encoding.UTF8.GetBytes(strPtr, str.Length + 1, buffer, bufferSize);
}
```